### PR TITLE
Add Github Actions

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
This patch adds 2 github actions one for pull-request and one for push (post-merge)

### CODE REVIEW

File: .github/workflows/push.yaml
Comment (elezar) [inline .github/workflows/push.yaml:161]: Comment is incorrect. There's a 10 second sleep.

File: .github/workflows/push.yaml
Comment (elezar) [inline .github/workflows/push.yaml:29]: Nit: multi-line YAML makes additions and removals easier to review.

Comment (ArangoGutierrez) [reply .github/workflows/push.yaml:29]: k

**Comment by ArangoGutierrez** [comment]
@cdesiniotis PTAL

File: .github/workflows/build.yaml
Comment (elezar) [inline .github/workflows/build.yaml:27]: See https://github.com/NVIDIA/libnvidia-container/pull/239 and https://github.com/NVIDIA/nvidia-container-toolkit/pull/332 for a way to selectively run entries in the parallel matrix on PRs. I would rather do that here too than maintain two separate workflows.

File: .github/workflows/push.yaml
Comment (elezar) [inline .github/workflows/push.yaml:131]: Question: Why not have `base` be a step in the precompiled chain instead of a separate job?

**Review by elezar — CHANGES_REQUESTED** [review]

File: .github/workflows/push.yaml
Comment (ArangoGutierrez) [reply .github/workflows/push.yaml:131]: I don't see why not, my initial intention was to directly translate the gilab pipeline, and optimize it later down the road. but I like the approach

File: .github/workflows/build.yaml
Comment (ArangoGutierrez) [reply .github/workflows/build.yaml:27]: I don't see why not, my initial intention was to directly translate the gilab pipeline, and optimize it later down the road. but I like the approach

File: .github/workflows/push.yaml
Comment (cdesiniotis) [inline .github/workflows/push.yaml:58]: Is this if-else block really necessary? I am not very familiar with Github actions, but doesn't the `on` block at the top of this file determine when this job runs? For example, as currently implemented, this job only runs on commits made to `main` and `release-*` branches, right?

Comment (ArangoGutierrez) [reply .github/workflows/push.yaml:58]: is a work around since we can not push during review process (from a fork) only once is merged (once source is in branch)

File: .github/workflows/image.yaml
Comment (elezar) [inline .github/workflows/image.yaml:173]: Nit: Let's drop this.

File: .github/workflows/image.yaml
Comment (elezar) [inline .github/workflows/image.yaml:158]: Should `-jammy` be `${BASE_TARGET}`?
```suggestion
          make DRIVER_BRANCH=${{ matrix.driver }} KERNEL_FLAVOR=${{ matrix.flavor }} build-base-${BASE_TARGET}
```

File: .github/workflows/image.yaml
Comment (elezar) [inline .github/workflows/image.yaml:157]: Question: Why are we not using the `precompiled.sh` script here?

File: .github/workflows/image.yaml
Comment (elezar) [inline .github/workflows/image.yaml:161]: I don't understand why this is needed. The `base/Dockerfile` has a sleep entrypoint. What does running it do?

If we need the file out of it we could run `cp` in a shell or even just `cat /var/kernel_version.txt` if mounts are an issue.

File: .github/workflows/image.yaml
Comment (elezar) [inline .github/workflows/image.yaml:175]: Does `kernel_version.txt` need to be "exported" from the previous step in some way?

File: .github/workflows/image.yaml
Comment (elezar) [inline .github/workflows/image.yaml:176]: Why don't we reuse the `precompiled.sh` script here?

File: .github/workflows/image.yaml
Comment (ArangoGutierrez) [reply .github/workflows/image.yaml:161]: well this was a work around for GitLab pipelines were Docker version and GitLab runner prevented us from doing what you say. so makes sense now that we have a bit more freedom to move to an updated way

File: .github/workflows/image.yaml
Comment (ArangoGutierrez) [reply .github/workflows/image.yaml:161]: that script was built to work around GitLab runner capabilities. once we move to Github it can be deprecated

File: .github/workflows/image.yaml
Comment (cdesiniotis) [inline .github/workflows/image.yaml:37]: Note, 470 is not supported on ubuntu22.04 and rhel9, so we need to skip this version for those distros. See https://gitlab.com/nvidia/container-images/driver/-/blob/main/.common-ci.yml?ref_type=heads#L78 and https://gitlab.com/nvidia/container-images/driver/-/blob/main/.common-ci.yml?ref_type=heads#L91

File: .github/workflows/image.yaml
Comment (cdesiniotis) [inline .github/workflows/image.yaml:55]: If I made a change to the `rhel8` folder in a PR, is there anyway to trigger the build for it?

File: .github/workflows/image.yaml
Comment (cdesiniotis) [inline .github/workflows/image.yaml:108]: Remove 550 from this matrix. The jobs will continue to fail since no precompiled packages exist for 550 yet.

File: .github/workflows/image.yaml
Comment (elezar) [reply .github/workflows/image.yaml:161]: OK. I suppose my question is whether a script makes sense at some point for users / customers to be able to build their own images.

Note a blocker in this PR though.

File: .github/workflows/image.yaml
Comment (elezar) [reply .github/workflows/image.yaml:55]: That's a good question. I think we could have excludes based on changed patterns.

File: .github/workflows/push.yaml
Comment (cdesiniotis) [reply .github/workflows/push.yaml]: Okay. Are you planning on updating this in a follow-up?

File: .github/workflows/image.yaml
Comment (cdesiniotis) [reply .github/workflows/image.yaml]: @ArangoGutierrez why was this question resolved?

**Review by cdesiniotis — APPROVED** [review]
Since the builds are working for 550 and precompiled, I am okay to merge this. We can address any follow-ups / improvements in a follow-up.

File: .github/workflows/image.yaml
Comment (ArangoGutierrez) [reply .github/workflows/image.yaml]: it should be a follow up

**Comment by ArangoGutierrez** [comment]
> Since the builds are working for 550 and precompiled, I am okay to merge this. We can address any follow-ups / improvements in a follow-up.

Also remember that gitlab pipelines will continue to work in the mean time, so we are double covered for now

**Review by elezar — APPROVED** [review]

File: .github/workflows/image.yaml
Comment (tariq1890) [inline .github/workflows/image.yaml:39]: We don't have this released yet. Might want to comment this out before merging

Comment (cdesiniotis) [reply .github/workflows/image.yaml:39]: It is okay to keep. We have not published it to nvcr.io yet, but it is valid to build 550 images in our GH actions.

Comment (tariq1890) [reply .github/workflows/image.yaml:39]: I thought we wait till the official release of the first 550 image, but it does make sense to build pre-release images of R550

**Review by tariq1890 — APPROVED** [review]

**Pull request merged**
